learning_journal/learning_journal/views/default.py

```diff
@@ -0,0 +1,74 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.httpexceptions import HTTPFound
+import datetime
+
+from sqlalchemy.exc import DBAPIError
+
+from ..models import MyEntry
+
+
+@view_config(route_name="home", renderer="../templates/home.jinja2")
+def home_view(request):
+    query = request.dbsession.query(MyEntry).order_by(MyEntry.creation_date.desc())
+    entries = query.all()
+    return {"entries": entries}
+
+
+@view_config(route_name='detail', renderer='../templates/detail.jinja2')
+def detail_view(request):
+    entry_id = int(request.matchdict['id'])
+    entry = request.dbsession.query(MyEntry).get(entry_id)
+    if entry is None:
+        raise HTTPNotFound
+    return {"entry": entry}
+
+
+@view_config(route_name='update', renderer='../templates/edit.jinja2')
+def edit_view(request):
+    entry_id = int(request.matchdict['id'])
+    entry = request.dbsession.query(MyEntry).get(entry_id)
+    if entry is None:
+        raise HTTPNotFound
+    if request.method == "POST":
+        entry.title = request.POST["title"]
+        entry.body = request.POST["body"]
+        entry = MyEntry(title=entry.title, body=entry.body, creation_date=entry.creation_date)
+        return HTTPFound(location=request.route_url('home'))
+    return {"entry": entry}
+
+
+@view_config(route_name='create', renderer='../templates/create.jinja2')
+def create_view(request):
+    title = body = error = ''
+    if request.method == 'POST':
```

Comment:
An additional question:  Why do you repeat the test for `request.method`?  Can you not combine the code in the two blocks into one block?